ibis/sql/sqlite/tests/test_functions.py

```diff
@@ -781,3 +781,36 @@ def test_scalar_parameter(db):
     result = expr.execute(params={start: start_string, end: end_string})
     expected = col.between(start_string, end_string).execute()
     tm.assert_series_equal(result, expected)
+
+
+def test_compile_twice(dbpath):
+    con1 = ibis.sqlite.connect(dbpath)
+    t1 = con1.table('batting')
+    sort_key1 = ibis.desc(t1.playerID)
+    sorted_table1 = t1.sort_by(sort_key1)
+    expr1 = sorted_table1.count()
+
+    con2 = ibis.sqlite.connect(dbpath)
+    t2 = con2.table('batting')
+    sort_key2 = ibis.desc(t2.playerID)
+    sorted_table2 = t2.sort_by(sort_key2)
+    expr2 = sorted_table2.count()
+
+    result1 = str(expr1.compile())
+    result2 = str(expr2.compile())
+
+    assert result1 == result2
+
+
+def test_count_on_order_by(db):
+    t = db.batting
+    sort_key = ibis.desc(t.playerID)
+    sorted_table = t.sort_by(sort_key)
+    expr = sorted_table.count()
+    result = str(
+        expr.compile().compile(compile_kwargs={'literal_binds': True})
+    )
+    expected = """\
+SELECT count('*') AS count 
```

Comment:
This is the count-star syntax for sqlite, I guess (did not know that)?

Reply:
Hm, actually this will have the same behavior as `count(*)` except in the case of an empty table. I need to fix this.

Reply:
Actually, nevermind. This is the form suggested by the SQLAlchemy docs.